### PR TITLE
Jroach/us103571 use it in qe

### PR DESCRIPTION
**Pull request by jeroach**
Use the new `d2l-activity-name` component in `d2l-evaluation-hub`

Also a bit of refactoring and cleanup around how we render the rows in `d2l-evaluation-hub-activities-list` (dropping the `hasLink` property since it's only used by one header; this seems cleaner IMO)

### CODE REVIEW

**Comment by jeroach** [comment]
@mtjandra @svanherk @CarolBusman Sorry, I got too excited and forgot to check if I broke tests :(

I'll update once I've fixed those. Sorry for the noise.

**Comment by jeroach** [comment]
@mtjandra @svanherk @CarolBusman Ready for review! Thanks!

Tests are green. The demo is not quite working, but I want to merge with Alex's recent demo test data changes to fix it.

Figure that can be a separate PR (unless people feel otherwise) :)

**Comment by jeroach** [comment]
After offline discussion - going to fix the demo data now that `master` was merged into the PR branch.

Stay tuned :) Review still appreciated on what's here. Thanks! @svanherk @CarolBusman @mtjandra

**Comment by jeroach** [comment]
Screenshot of demo:
quick_eval_demo

File: demo/d2l-evaluation-hub/demo-data-parser.js [hunk 1]
```diff
@@ -438,7 +446,24 @@ function getMappings(data) {
 		mappings[getHrefForCourseId(i)] = formatCourse(course, getHrefForEnrollments(i));
 	});
 
-	const pagedActivities = chunk(activities.map(a => formatActivity(a)), 3);
+	const pagedActivities = [];
+	let activitiesOnPage = [];
```
Comment (jeroach) [inline demo/d2l-evaluation-hub/demo-data-parser.js]: @AlexBedley I wrote this to enable setting the activity route with an id. I removed the `chunk` call and import.

Please let me know if you have any concerns. Thanks!

**Comment by svanherk** [comment]
Not really related to the hookup, more the UI component underneath - Does Erin want the text to wrap under the icon, or for the icon to be to the left and vertically centered?

**Comment by jeroach** [comment]
> Not really related to the hookup, more the UI component underneath - Does Erin want the text to wrap under the icon, or for the icon to be to the left and vertically centered?

Good question - I'll follow up with Erin.

File: components/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.js [hunk 2]
```diff
@@ -238,21 +239,20 @@ class D2LEvaluationHubActivitiesList extends mixinBehaviors([D2L.PolymerBehavior
 				var item = {
 					displayName: '',
 					courseName: '',
-					activityName: '',
+					activityName: this._getActivityNameHref(activity),
```
Comment (svanherk) [inline components/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.js]: Minor: Should we update this property to `activityNameHref` to more accurately reflect the data it holds now?

File: test/d2l-evaluation-hub/data/unassessedActivities.json [hunk 3]
```diff
@@ -10,6 +10,10 @@
 			"class": ["user-assignment-activity","activity"],
 			"rel":["https://activities.api.brightspace.com/rels/user-activity-usage"],
 			"links":[
+        {
```
Comment (svanherk) [inline test/d2l-evaluation-hub/data/unassessedActivities.json]: Indenting is weird here

File: demo/d2l-evaluation-hub/demo-data-parser.js (hunk 1 above)
Comment (jeroach) [reply demo/d2l-evaluation-hub/demo-data-parser.js]: And.....merge conflicts :)

@AlexBedley I'll work on moving/refactoring my logic to move the code over into `page-handler.js`

If you have any thought/concerns with my approach, please let me know.

File: test/d2l-evaluation-hub/data/unassessedActivities.json (hunk 3 above)
Comment (jeroach) [reply test/d2l-evaluation-hub/data/unassessedActivities.json]: I'll take a look

**Comment by jeroach** [comment]
Code is in a weird state right now. I resolved some merge conflicts such that the code complies, but the logic isn't fully resolved.

Working on that currently.

File: demo/d2l-evaluation-hub/demo-data-parser.js (hunk 1 above)
Comment (jeroach) [reply demo/d2l-evaluation-hub/demo-data-parser.js]: @AlexBedley Conflicts resolved :)

Please let me know if you have thoughts/concerns with my approach. Thanks!

**Comment by jeroach** [comment]
@svanherk @CarolBusman @mtjandra @AlexBedley PR comments/merge conflicts are addressed. Ready for final review! Thanks!

**Comment by jeroach** [comment]
As per offline discussion with Alex, I'll be making some changes to the logic (as sorting will not work as I have it now).

File: demo/d2l-evaluation-hub/activity-handler.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+		formattedActivity.entities.push(draftEntity);
+	}
+
+	return formattedActivity;
+}
+
+function formatActivities(activities) {
+	const formattedActivities = [];
+
+	activities.forEach((activity, i) => {
```
Comment (AlexBedley) [inline demo/d2l-evaluation-hub/activity-handler.js]: nit: `map` would be more idiomatic here like: `const formattedActivities = activities.map(...)`

Comment (jeroach) [reply demo/d2l-evaluation-hub/activity-handler.js]: Fixed

**Pull request merged**